File: temporal-sdk/src/main/java/io/temporal/internal/sync/SyncReplayWorkflowFactory.java
package io.temporal.internal.sync;

import io.temporal.common.context.ContextPropagator;
import io.temporal.common.converter.DataConverter;
import io.temporal.internal.replay.ReplayWorkflow;
import io.temporal.worker.WorkflowImplementationOptions;
import java.util.List;

public interface SyncReplayWorkflowFactory {

  ReplayWorkflow getWorkflow(
      SyncWorkflowDefinition workflow,
      WorkflowImplementationOptions options,
      DataConverter dataConverter,
      List<ContextPropagator> contextPropagators
  );
}

File: temporal-kotlin/src/main/kotlin/io/temporal/internal/sync/SyncReplayWorkflowFactory.kt
package io.temporal.internal.sync

import io.temporal.common.context.ContextPropagator
import io.temporal.common.converter.DataConverter
import io.temporal.internal.replay.ReplayWorkflow
import io.temporal.worker.WorkflowImplementationOptions

internal interface SyncReplayWorkflowFactory {

  fun getWorkflow(
    workflow: SyncWorkflowDefinition,
    options: WorkflowImplementationOptions?,
    dataConverter: DataConverter,
    contextPropagators: List<ContextPropagator>
  ): ReplayWorkflow
}